tests/integration/test_authd/test_authd_key_request_exec_path.py

```diff
@@ -0,0 +1,170 @@
+log_monitor_paths = [LOG_FILE_PATH]
+receiver_sockets_params = [(kreq_sock_path, 'AF_UNIX', 'UDP')]
+test_case_ids = [f"{test_case['name'].lower().replace(' ', '-')}" for test_case in message_tests]
+
+monitored_sockets_params = [('wazuh-authd', None, True)]
+receiver_sockets, monitored_sockets, log_monitors = None, None, None  # Set in the fixtures
+
+# Tests
+test_index = 0
+
+
+def get_current_test():
+    """
+    Get the current test case.
+    """
+    global test_index
+    current = test_index
+    test_index += 1
+    return current
+
+
+@pytest.fixture(scope="module", params=configurations, ids=test_case_ids)
+def get_configuration(request):
+    """Get configurations from the module"""
+    yield request.param
+
+
+def test_key_request_exec_path(get_configuration, configure_local_internal_options_module, configure_environment,
+                               configure_sockets_environment, connect_to_sockets_function, tear_down):
+    '''
+    description:
+        Checks that every input message on the key request port with different exec_path configuration
+        shows the corresponding error in the manager logs.
+
+    wazuh_min_version: 4.4.0
+
+    parameters:
+        - get_configuration:
+            type: fixture
+            brief: Get the configuration of the test.
+        - configure_environment:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - configure_local_internal_options_module:
+            type: fixture
+            brief: Configure the local internal options file.
+        - configure_sockets_environment_function:
+            type: fixture
+            brief: Configure the socket listener to receive and send messages on the sockets at function scope.
+        - connect_to_sockets_function:
+            type: fixture
+            brief: Bind to the configured sockets at function scope.
+        - tear_down:
+            type: fixture
+            brief: Cleans the client.keys file.
+
+    assertions:
+        - The exec_path must be configured correctly
+        - The script works as expected
+
+    input_description:
+        Different test cases are contained in an external YAML file (test_key_request_exec_path.yaml) which
+        includes the different possible key requests with different configurations and the expected responses.
+
+    expected_log:
+        - Key request responses on 'authd' logs.
+    '''
+    current_test = get_current_test()
+    case = message_tests[current_test]['test_case']
+    # Overwrites the configuration with the values ​​set in the YAML file
+    override_wazuh_conf(get_configuration)
+    key_request_sock = receiver_sockets[0]
+
+    for stage in case:
+        # Reopen socket (socket is closed by manager after sending message with client key)
+        key_request_sock.open()
+        message = stage['input']
+        key_request_sock.send(message, size=False)
```

Comment:
The input message should be generated with the attributes `ip` and `id` set in the yaml file.